topostats/processing.py

```diff
@@ -249,10 +280,26 @@ def process_scan(
                     # Set tracing_stats_df in light of direction
                     if grains_config["direction"] == "both":
                         grainstats_df = pd.concat([grainstats["below"], grainstats["above"]])
+                        image_stats["grains_number_above"] = grainstats["above"].shape[0]
```

Comment:
Here each of the `if:...elif:` statements is doing the same thing, the only thing that differs between them is what is being put in.

This makes it the perfect candidate for a function. But thinking about the `roughness.py` module would this functionality not fit better in the `Grainstats` class as `processing.py` is more about what methods to call and when.